examples/experimental/tensorflow/classification/main.py

```diff
@@ -0,0 +1,232 @@
+    callbacks.extend(compression_callbacks)
+
+    validation_kwargs = {
+        'validation_data': validation_dataset,
+        'validation_steps': validation_steps,
+        'validation_freq': config.test_every_n_epochs,
+    }
+
+    if 'train' in config.mode:
+        if is_accuracy_aware_training(config):
+            logger.info('starting an accuracy-aware training loop...')
+            result_dict_to_val_metric_fn = lambda results: 100 * results['acc@1']
+            compress_model.accuracy_aware_fit(train_dataset,
+                                              compression_ctrl,
+                                              nncf_config=config.nncf_config,
+                                              callbacks=callbacks,
+                                              initial_epoch=initial_epoch,
+                                              steps_per_epoch=train_steps,
+                                              tensorboard_writer=SummaryWriter(config.log_dir,
+                                                                               'accuracy_aware_training'),
+                                              log_dir=config.log_dir,
+                                              uncompressed_model_accuracy=uncompressed_model_accuracy,
+                                              result_dict_to_val_metric_fn=result_dict_to_val_metric_fn,
+                                              **validation_kwargs)
+        else:
+            logger.info('training...')
+            compress_model.fit(
+                train_dataset,
+                epochs=train_epochs,
+                steps_per_epoch=train_steps,
+                initial_epoch=initial_epoch,
+                callbacks=callbacks,
+                **validation_kwargs)
+
+    logger.info('evaluation...')
+    statistics = compression_ctrl.statistics()
+    logger.info(statistics.to_str())
+    results = compress_model.evaluate(
+        validation_dataset,
+        steps=validation_steps,
+        callbacks=[get_progress_bar(
+            stateful_metrics=['loss'] + [metric.name for metric in metrics])],
+        verbose=1)
+
+    if config.metrics_dump is not None:
+        write_metrics(results[1], config.metrics_dump)
+
+    if 'export' in config.mode:
+        save_path, save_format = get_saving_parameters(config)
+        compression_ctrl.export_model(save_path, save_format)
+        logger.info('Saved to {}'.format(save_path))
+
+
+def export(config):
+    pass
+
+
+def main(argv):
+    parser = get_argument_parser()
+    config = get_config_from_argv(argv, parser)
+    print_args(config)
+
+    serialize_config(config.nncf_config, config.log_dir)
+    serialize_cli_args(parser, argv, config.log_dir)
+
+    nncf_root = Path(__file__).absolute().parents[3]
+    create_code_snapshot(nncf_root, osp.join(config.log_dir, 'snapshot.tar.gz'))
+    if 'train' in config.mode or 'test' in config.mode:
+        run(config)
+    elif 'export' in config.mode:
```

Comment:
why do we have logic for the export in 2 places: in main and run methods?

Reply:
In the examples from the develop branch we use the same approach. Sometimes we want to export existing checkpoints to some format (frozen graph, saved model, h5). For this task, we use `--mode export` param.